FSW/src/util/hammingBlock.cpp

```diff
@@ -0,0 +1,269 @@
+/* hammingBlock.cpp defines the HammingBlock class
+ * Usage:
+ *  A HammingBlock object is a container for an 8 byte message, which is encoded into a 9 byte
+ *  block using a (72,64) Hamming code.
+ *  A HammingBlock can encode/decode itself, scan its contetns for errors, and correct single bit errors.
+ * 
+ * Modules encompassed:
+ *  Memory Scrubbing
+ *  Program EDAC
+ *
+ * Additional files needed for compilation:
+ *  config.hpp
+ *  
+ */
+
+/* - - - - - - Includes - - - - - - */
+// All libraries are put in edac.hpp
+// NS2 headers
+#include "../headers/hammingBlock.hpp"
+
+
+/* - - - - - - HammingBlock Constructor - - - - - - *
+* Inputs:
+*   None
+*/
+HammingBlock::HammingBlock() {
+    memset(m_block, 0, HAMMING_BLOCK_SIZE);
+}
+
+
+/* - - - - - - encodeMessage - - - - - - *
+ * Usage:
+ *  Encodes an 8 byte message into the block
+ * 
+ * Inputs:
+ *  message - pointer to a single 8 byte message block, typecast to void*
+ *  
+ * Outputs:
+ *  None
+ */
+void HammingBlock::encodeMessage(void *message) {
```

Comment:
More straightforward than I would have expected. Going to trust you on the algorithm, but this is a perfect example case for where unit testing really helps out, would go nicely in one of the final reports.

Reply:
Yeah, Hamming codes are super elegant to implement thanks to bitwise operations.
Most of the magic happens in `indexParity ^= hammingIdx`